modules/imaging_uploader/jsx/UploadForm.js

```diff
@@ -77,16 +77,42 @@ class UploadForm extends React.Component {
   submitForm() {
     // Validate required fields
     const data = this.state.formData;
-    if (!data.mri_file || !data.IsPhantom) {
+
+    if (!data.mriFile || !data.IsPhantom) {
       return;
     }
 
+    const fileName = data.mriFile.name;
+    if (data.IsPhantom === 'N') {
+      if (!data.candID || !data.pSCID || !data.visitLabel) {
+        return;
+      }
+      // Make sure file follows PSCID_CandID_VL[_*].zip|.tgz|.tar.gz format
+      const pcv = data.pSCID + '_' + data.candID + '_' + data.visitLabel;
+      const pcvu = pcv + '_';
+      const properName = new RegExp("^" + pcv + ".(zip|tgz|tar.gz)");
+      const properNameExt = new RegExp("^" + pcvu + ".*(.(zip|tgz|tar.gz))");
+      if (!fileName.match(properName) && !fileName.match(properNameExt)) {
+        swal({
+          title: "File improperly named!",
+          text: "File name must match " + pcv +
```

Comment:
I would rephrase the error message to the following:
"File name and values in the PSCID, CandID and Visit label fields of the instrument do not match. Please verify that the information entered in the fields or the file name are correct."

I find it dangerous to suggest them that it is the file name that is not correct (it is possible that what is not correct is one of the field instead of the filename)